src/lib/chips/chip-list.ts

```diff
@@ -15,13 +15,30 @@ import {
   QueryList,
   ViewEncapsulation,
   OnDestroy,
+  Optional,
+  ElementRef,
+  Renderer2,
 } from '@angular/core';
 
 import {MdChip} from './chip';
 import {FocusKeyManager} from '../core/a11y/focus-key-manager';
-import {SPACE, LEFT_ARROW, RIGHT_ARROW, TAB} from '../core/keyboard/keycodes';
 import {coerceBooleanProperty} from '@angular/cdk';
 import {Subscription} from 'rxjs/Subscription';
+import {
+  LEFT_ARROW, RIGHT_ARROW, BACKSPACE, DELETE, UP_ARROW, DOWN_ARROW
+} from '../core/keyboard/keycodes';
+import {Directionality} from '../core';
+
+/** Utility to check if an input element has no value. */
+function _isInputEmpty(element: HTMLElement): boolean {
```

Comment:
Doesn't need to be prefixed with an underscore (since it's internal to the module regardless)